Common/include/toolboxes/C1DInterpolation.hpp

```diff
@@ -0,0 +1,167 @@
+    int n; /*!< \brief local variable for holding the number of intervals */
+public:
+    
+    /*!
+    * \brief Constructor of the CAkimaInterpolation class.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    CAkimaInterpolation(vector<su2double> &X, vector<su2double> &Data){
+        SetSpline(X,Data);
+    }
+
+    /*!
+    * \brief Destructor of the CAkimaInterpolation class.
+    */
+    ~CAkimaInterpolation(){}
+
+    /*!
+    * \brief for setting the cofficients for the Akima spline.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    void SetSpline(vector<su2double> &x, vector<su2double> &y) override;
+
+    /*!
+    * \brief For evaluating the value of Akima Spline.
+    * \param[in] Point_Interp - the point where interpolation value is required.
+    * \returns the interpolated value.
+    */
+    su2double EvaluateSpline(su2double Point_Interp) override;
+};
+
+class CLinearInterpolation: public C1DInterpolation{
+    protected:
+    vector<su2double> x,y,dydx; /*!< \brief local variables for linear 'spline' cooefficients */
+    public:
+    
+    /*!
+    * \brief Constructor of the CLinearInterpolation class.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    CLinearInterpolation(vector<su2double> &X, vector<su2double> &Data){
+        SetSpline(X,Data);
+    }
+
+    /*!
+    * \brief Destructor of the CInletInterpolation class.
+    */
+    ~CLinearInterpolation(){}
+
+    /*!
+    * \brief for setting the cofficients for Linear 'spline'.
+    * \param[in] X - the x values.
+    * \param[in] Data - the f(x) values.
+    */
+    void SetSpline(vector<su2double> &x, vector<su2double> &y) override;
+
+    /*!
+    * \brief For evaluating the value for Linear 'spline'.
+    * \param[in] Point_Interp - the point where interpolation value is required.
+    * \returns the interpolated value.
+    */
+    su2double EvaluateSpline(su2double Point_Interp) override;
+};
+
+/*!
+* \brief to correct for interpolation type.
+* \param[in] Inlet_Interpolated - the interpolated data after spline evaluation.
+* \param[in] Theta - the angle of the vertex (in xy plane).
+* \param[in] nDim - the dimensions of the case.
+* \param[in] Coord - the coordinates of the vertex.
+* \param[in] nVar_Turb - the number of turbulence variables as defined by turbulence model
+* \param[in] *config - Definition of the particular problem.
+* \returns the corrected Inlet Interpolated Data.
+*/
+vector<su2double> CorrectedInletValues(vector<su2double> &Inlet_Interpolated, 
+                                    su2double Theta ,
+                                    unsigned short nDim, 
+                                    su2double *Coord, 
+                                    unsigned short nVar_Turb, 
+                                    CConfig *config);
```

Comment:
Ah yes thank you, this was in one of my older comments :+1:

Reply:
I would want to do that too, but then I would be passing an unsigned short (enum), can I check that neatly with switch without #include<config>?

Reply:
@talbring you need to define the radius at which you want those values to be calculated. For that, it makes sense to have a Vr_Vtheta as default (think like cylindrical coordinates). I can include an option for Vx_Vy for defining at a particular radius if you think that might be useful.

Reply:
Include option_structure.hpp instead of CConfig.hpp

Reply:
And do not use unsigned short, use the actual enum type

Reply:
@pcarruscag also, I am printing 'Inlet Interpolation being done using AKIMA function and type ALPHA_PHI for inlet' during execution in the main LoadInletProfile function and it seems to appear twice. How can I avoid that? I think it is being called from CDriver twice on lines 1409-1413.

Reply:
And if you run on 4 cores does it print 4 times? If so you need
```
if (rank == MASTER_NODE) {
  // your print statements
}
```
If it is printed 2 times because it is called for flow and turbulence variables I guess it is fine...

Reply:
@pcarruscag please be a little more specific about how to pass the enum options. I am doing this but it is not working:

`              switch(config->GetKindInletInterpolationType()){
              case(VR_VTHETA):
                Inlet_Values = CorrectedInletValues(Inlet_Interpolated, Theta, nDim, Coord, 
nVar_Turb, VR_VTHETA);
              break;
              case(ALPHA_PHI):
                Inlet_Values = CorrectedInletValues(Inlet_Interpolated, Theta, nDim, Coord, nVar_Turb, ALPHA_PHI);
              break;
              }`

```
vector<su2double> CorrectedInletValues(.....enum ENUM_INLET_SPANWISEINTERPOLATION)
```

I also tried passing `config->GetKindInletInterpolationType()` but it didnt work

Reply:
Its not dependent on the number of cores. When I run Euler it only prints once.